components/automate-cli/cmd/chef-automate/start_linx.go

```diff
@@ -1,67 +1,203 @@
+		newSSHConfig := &SSHConfig{
+			sshUser:    originalSSHConfig.sshUser,
+			sshPort:    originalSSHConfig.sshPort,
+			sshKeyFile: originalSSHConfig.sshKeyFile,
+			hostIP:     hostIP,
 		}
+		newSSHUtil := NewSSHUtil(newSSHConfig)
+
+		go func(args []string, remoteService string, scriptCommands string, newSSHUtil SSHUtil, resultChan chan ResultConfigSet) {
+			rc := ResultConfigSet{newSSHUtil.getSSHConfig().hostIP, "", nil}
+			output, err := newSSHUtil.connectAndExecuteCommandOnRemote(scriptCommands, true)
+			if err != nil {
+				rc.Error = err
+				resultChan <- rc
+				return
+			}
+
+			err = checkOutputForError(output)
+			if err != nil {
+				rc.Error = err
+				resultChan <- rc
+				return
+			}
+
+			rc.Output = output
+			resultChan <- rc
+		}(args, remoteService, scriptCommands, newSSHUtil, resultChan)
 	}
 
+	for i := 0; i < len(Ips); i++ {
+		result := <-resultChan
+
+		if i == 0 {
+			writer.StopSpinner()
+		}
+		printStartConnectionMessage(remoteService, result.HostIP, cliWriter)
+		if result.Error != nil {
+			printStartErrorMessage(remoteService, result.HostIP, cliWriter, result.Error)
+		} else {
+			cliWriter.Printf("Output for Host IP %s : %s", result.HostIP, result.Output+"\n")
+			printStartSuccessMessage(remoteService, result.HostIP, cliWriter)
+		}
+
+		if i < len(Ips)-1 {
+			writer.StartSpinner()
+		}
+	}
+
+	close(resultChan)
 	return nil
 }
 
-func init() {
-	RootCmd.AddCommand(startCommand)
+func checkNodes(args []string, sshUtil SSHUtil, Ips []string, remoteService string, cliWriter *cli.Writer) error {
+	if len(Ips) == 0 {
+		writer.Errorf("No %s IPs are found", remoteService)
+		return status.Errorf(1, "No %s IPs are found", remoteService)
+	}
+	var err error
+	if isManagedServicesOn() {
+		return status.Errorf(status.InvalidCommandArgsError, ERROR_SELF_MANAGED_START, remoteService)
+	}
+	if remoteService == "opensearch" || remoteService == "postgresql" {
+		for i := 0; i < len(Ips); i++ {
+			err = startBackEndNodes(args, sshUtil, Ips[i], remoteService, cliWriter)
+		}
+	} else {
+		err = startFrontEndNodes(args, sshUtil, Ips, remoteService, cliWriter)
+	}
+	return err
+}
+
+func startBackEndNodes(args []string, sshUtil SSHUtil, Ips string, remoteService string, cliWriter *cli.Writer) error {
+	sshUtil.getSSHConfig().hostIP = Ips
+	scriptCommands := "sudo systemctl start hab-sup"
+	printStartConnectionMessage(remoteService, sshUtil.getSSHConfig().hostIP, writer)
+	_, err := sshUtil.connectAndExecuteCommandOnRemote(`sudo hab license accept; sudo hab svc status`, true)
+	if err != nil {
+		printStartErrorMessage(remoteService, Ips, cliWriter, err)
+		return err
+	}
+
+	output, err := sshUtil.connectAndExecuteCommandOnRemote(scriptCommands, true)
+	if err != nil {
+		cliWriter.Errorf("%v\n", err)
+		return err
+	}
+
+	err = checkOutputForError(output)
+	if err != nil {
+		cliWriter.Errorf("%v\n", err)
+		return err
+	}
+
+	if err != nil {
+		printStartErrorMessage(remoteService, Ips, cliWriter, err)
```

Comment:
Should return the error too
```suggestion
		printStartErrorMessage(remoteService, Ips, cliWriter, err)
		return err
```

Reply:
done